UniWatch/Controllers/ClassController.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Web;
+using System.Web.Mvc;
+using UniWatch.DataAccess;
+using UniWatch.Models;
+
+namespace UniWatch.Controllers
+{
+    public class ClassController : Controller
+    {
+        private IDataAccess _dataAccess;
+
+        public ClassController()
+        {
+            _dataAccess = new DataAccess.DataAccess();
+        }
+
+        public ClassController(IDataAccess dataAccess)
+        {
+            _dataAccess = dataAccess;
+        }
+
+        //GET: Index
+        /// <summary>
+        /// Displays all classes taught by the teacher
+        /// </summary>
+        /// <param name="teacherId">Id of the teacher</param>
+        public ActionResult Index(int teacherId)
+        {
+            ViewBag.TeacherId = teacherId;
+            return View(_dataAccess.ClassManager.GetClassesForTeacher(teacherId));
+        }
+
+
+        // GET: Create
+        /// <summary>
+        /// Sends user to the create class page
+        /// </summary>
+        /// <param name="teacherId">Id of the teacher</param>
+        public ActionResult Create(int teacherId)
+        {
+            var teacher = _dataAccess.UserManager.GetTeacherById(teacherId);
+            return View(new Class() { Teacher = teacher });
+        }
+
+        //POST: Create
+        /// <summary>
+        /// Create class for the teacher
+        /// </summary>
+        /// <param name="class">Model containing information to create the class</param>
+        /// <param name="teacherId">Id of the teacher</param>
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public ActionResult Create([Bind(Include = "Name,Number,Section,Semester,Year")] Class @class, int teacherId)
+        {
+            if (!ModelState.IsValid)
+            {
+                return View(@class);
+            }
+
+            var created = _dataAccess.ClassManager.CreateClass(@class.Name, @class.Number, @class.Section, @class.Semester, @class.Year, teacherId);
```

Comment:
Surround this with a `try-catch` statement since `ClassManager.CreateClass` throws an `InvalidOperationException` if there is no teacher with that id or a duplicate class exists.